### ISSUE TITLE
multi-GPU

### ISSUE THREAD

**Issue by qizhust** [Vision-CAIR/VisualGPT#1]
Hi, thanks for the code! Is there a version supporting multi-gpu training?

**Comment by junchen14** [comment]
Thanks for asking question about our code.
our code does not support multi-gpu training at the moment, since our model is designed in a data efficient setting, and it does not consume a lot of time for training.

**Comment by junchen14** [comment]
> Hi, thanks for the code! Is there a version supporting multi-gpu training?
Thanks for asking question about our code.
our code does not support multi-gpu training at the moment, since our model is designed in a data efficient setting, and it does not consume a lot of time for training.

**Comment by junchen14** [comment]
> Hi, thanks for the code! Is there a version supporting multi-gpu training?

Thanks for asking question about our code.
our code does not support multi-gpu training at the moment, since our model is designed in a data efficient setting, and it does not consume a lot of time for training.

**Comment by qizhust** [comment]
Yep, but there might be OOM problem.

**Comment by junchen14** [comment]
sorry for the late reply.
yeah we have not implemented the multi-gpu training. 
There are several solutions to reduce the heavy gpu memory requeriment on single GPU
1.  use the lower batch size and accumulate the gradients in multiple steps.
2. You can adapt the pytorch multi-gpu training or other external tools such as accelerate for the deployment of multi-gpu training.

hope it will work for you

**Comment by qizhust** [comment]
Thanks!
